### PR TITLE
check for existing group name for create and update operations

### PR DESCRIPTION
**Pull request by rahulsayini**

### CODE REVIEW

File: jwala-services/src/main/java/com/cerner/jwala/service/group/impl/GroupServiceImpl.java [hunk 1]
```diff
@@ -65,7 +67,14 @@ public GroupServiceImpl(final GroupPersistenceService groupPersistenceService,
...
-
+        try {
+            groupPersistenceService.getGroup(createGroupRequest.getGroupName());
+            String message = MessageFormat.format("Group Name already exists: {0} ", createGroupRequest.getGroupName());
+            LOGGER.error(message);
+            throw new EntityExistsException(message);
+        } catch (NotFoundException e) {
+            LOGGER.debug("No group name conflict, ignoring not found exception for creating group ", e);
+        }
         createGroupRequest.validate();
```
Comment (jeddcuison) [inline jwala-services/src/main/java/com/cerner/jwala/service/group/impl/GroupServiceImpl.java]: Does validate check if the group name is empty or has illegal characters ? If so, createGroupRequest.validate() should go first before you check for duplicates.

File: jwala-services/src/main/java/com/cerner/jwala/service/group/impl/GroupServiceImpl.java [hunk 2]
```diff
@@ -112,6 +121,16 @@ public Group getGroup(final String name) {
...
+        try {
+            if (!orginalGroup.getName().equalsIgnoreCase(anUpdateGroupRequest.getNewName()) && null != groupPersistenceService.getGroup(anUpdateGroupRequest.getNewName())) {
+                String message = MessageFormat.format("Group Name already exists: {0}", anUpdateGroupRequest.getNewName());
+                LOGGER.error(message);
+                throw new EntityExistsException(message);
+            }
+        } catch (NotFoundException e) {
+            LOGGER.debug("No group name conflict, ignoring not found exception for creating group ", e);
+        }
         anUpdateGroupRequest.validate();
```
Comment (jeddcuison) [inline jwala-services/src/main/java/com/cerner/jwala/service/group/impl/GroupServiceImpl.java]: If validate checks if the name is legal e.g. is null, has illegal characters, length is good etc then validate should go first before checking for duplicates

File: jwala-services/src/test/java/com/cerner/jwala/service/group/impl/GroupServiceImplVerifyTest.java [hunk 3]
```diff
@@ -59,17 +61,28 @@ public GroupServiceImplVerifyTest() {
...
     @Test
     public void testCreateGroup() {
 
-        final CreateGroupRequest command = mock(CreateGroupRequest.class);
+        final CreateGroupRequest command = new CreateGroupRequest("Group");
+        when(Config.mockGroupPersistenceService.getGroup(anyString())).thenThrow(NotFoundException.class);
+        groupService.createGroup(command,
+                user);
+
+        verify(Config.mockGroupPersistenceService, times(1)).createGroup(command);
```
Comment (jeddcuison) [inline jwala-services/src/test/java/com/cerner/jwala/service/group/impl/GroupServiceImplVerifyTest.java:75]: This is minor, just an FYI - verify's default Times is 1 so you can omit it in the future

    verify(someMock).doSomething(...)

**Review by jeddcuison — COMMENTED** [review]
If validate checks if group name is empty, has illegal characters or has invalid length then it should execute first before checking for duplicates

**Review by PeterTeunissen — APPROVED** [review]

**Review by jmahmoodcc — APPROVED** [review]

**Pull request merged**